lib/lz4.c

```diff
@@ -1085,7 +1085,9 @@ LZ4_FORCE_INLINE int LZ4_compress_generic_validated(
 
         /* Catch up */
         filledIp = ip;
-        while (((ip>anchor) & (match > lowLimit)) && (unlikely(ip[-1]==match[-1]))) { ip--; match--; }
+        if ((match > lowLimit) && unlikely(ip[-1] == match[-1])) {
```

Comment:
We should probably add an `assert(ip > anchor)` before this line.
I understand this condition is supposed to be correct, it's just not "obvious enough" on reading this line.

Reply:
Thanks for that. Updated in the new version.